### ISSUE TITLE
Ignore file (`.dockerignore`) does not parse leading slash correctly

### ISSUE THREAD

**Issue by patarapolw** [containers/podman#11040]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

The old `.dockerignore` I used with docker-compose.yml doesn't work.

```ignore
*

!/packages/server/assets/
!/packages/server/scripts/
!/packages/server/public/
!/packages/server/src/
!/packages/server/package.json
!/packages/server/yarn.lock
!/packages/server/tsconfig.json
!/packages/server/requirements.txt
```

But it works after some correction.

```ignore
*

!packages/server/assets/
!packages/server/scripts/
!packages/server/public/
!packages/server/src/
!packages/server/package.json
!packages/server/yarn.lock
!packages/server/tsconfig.json
!packages/server/requirements.txt
```

Now, I am curious on how podman parse ignore files.

Also, normally, I would choose to include files, rather the ignore many things.

**Steps to reproduce the issue:**

1. Clone `git clone -b docker https://github.com/patarapolw/zhres.git`

2. `cd zhres && podman-compose build`

**Describe the results you received:**

```
STEP 9: COPY packages/server/package.json packages/server/yarn.lock /server/
Error: error building at STEP "COPY packages/server/package.json packages/server/yarn.lock /server/": no items matching glob "/home/polv/shared/linux/projects/zhres/packages/server/package.json" copied (1 filtered out): no such file or directory
```

**Describe the results you expected:**

No errors.

**Output of `podman version`:**

```
podman version 3.2.2
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.21.0
  cgroupControllers: []
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: /usr/bin/conmon is owned by conmon 1:2.0.29-1
    path: /usr/bin/conmon
    version: 'conmon version 2.0.29, commit: 7e6de6678f6ed8a18661e1d5721b81ccee293b9b'
  cpus: 8
  distribution:
    distribution: arch
    version: unknown
  eventLogger: journald
  hostname: polv-arch
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 998
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65537
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65537
  kernel: 5.13.4-arch2-1
  linkmode: dynamic
  memFree: 2010882048
  memTotal: 12077694976
  ociRuntime:
    name: crun
    package: /usr/bin/crun is owned by crun 0.20.1-2
    path: /usr/bin/crun
    version: |-
      crun version 0.20.1
      commit: 38271d1c8d9641a2cdc70acfa3dcb6996d124b3d
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /etc/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: /usr/bin/slirp4netns is owned by slirp4netns 1.1.11-1
    version: |-
      slirp4netns version 1.1.11
      commit: 368e69ccc074628d17a9bb9a35b8f4b9f74db4c6
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 8537219072
  swapTotal: 8589930496
  uptime: 17h 36m 48.59s (Approximately 0.71 days)
registries:
  search:
  - docker.io
store:
  configFile: /home/polv/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/polv/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 27
  runRoot: /run/user/1000/containers
  volumePath: /home/polv/.local/share/containers/storage/volumes
version:
  APIVersion: 3.2.2
  Built: 1625835244
  BuiltTime: Fri Jul  9 19:54:04 2021
  GitCommit: d577c44e359f9f8284b38cf984f939b3020badc3
  GoVersion: go1.16.5
  OsArch: linux/amd64
  Version: 3.2.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
➜  ~ pacman -Qs podman
local/podman 3.2.2-2
    Tool and library for running OCI-based containers in pods
local/podman-compose 0.1.5-2
    A script to run docker-compose.yml using podman
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

Yes

**Comment by rhatdan** [comment]
Fixed in 36e17cbb9bfa6cc30a23f81097902e18e1af0ae8

So this is fixed in the main branch right now.

**Comment by rhatdan** [comment]
We plan on shipping podman 3.3 rc1 this week, which will have the fix.
